src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1604,7 +1604,8 @@ abstract class RefChecks extends Transform {
         // this may be the implementation of e.g. a generic method being parameterized
         // on Unit, in which case we had better let it slide.
         val isOk = (
-             sym.isGetter
+          sym.isSuperAccessor
+          || sym.isGetter
```

Comment:
```
        val isOk = (
             sym.isGetter
          || sym.isDefaultGetter
          || sym.allOverriddenSymbols.exists(over => !(over.tpe.resultType =:= sym.tpe.resultType))
          || sym.isArtifact
        )
```
I did not investigate what else is caught by `isArtifact`. Putting the rare case last in short-circuiting boolean expression is always a good idea; though it is the cheap test here. Also keep existing code format style, which is idiosyncratic here.

Reply:
Yeah, `isArtifact` seems more suitable.